docker-compose.yml

```diff
@@ -0,0 +1,66 @@
+version: "3.1"
+
+services:
+    db:
+        image: timescale/timescaledb:latest-pg16
+        environment:
+            POSTGRES_USER: postgres
+            POSTGRES_PASSWORD: postgres
+
+        volumes:
+            - ./data:/var/lib/postgresql/data
+        ports:
+            - "5432:5432"
+        healthcheck:
+            test: ["CMD", "pg_isready", "-U", "postgres", "-d", "postgres"]
+            interval: 1s
+            timeout: 1s
+            retries: 25
+
+    db-setup:
```

Comment:
So this is a "command" in the form of a service. You can probably use "docker exec" which would let you run a command inside the container and your setup command won't need to be a service anymore.

Reply:
Oh thanks for the feedback! Uhm... could you elaborate a bit how I could implement this alternative? :sweat_smile: 

Btw I took this answer as inspiration for the db-setup service in the docker-compose: https://stackoverflow.com/a/64153124

Reply:
Yeah ofc! 🙃

https://docs.docker.com/reference/cli/docker/container/exec/

U can use this to run any command inside the container. Then u can call ur bash script to setup the db.

I'd personally do "npm run setup" calling "docker exec" calling the bash script to set up the stuff. 

You have to either copy the bash script to the container in your dockerfile or use mapped volumes. 

I would say since we're not that big yet use mapped volumes so basically you'd have direct access of your "host" files inside the container. (This is only good for dev environments) If we were to deploy this to production we'd probably just use COPY in dockerfile that's why people often have environment-centric dockerfiles.

https://docs.docker.com/storage/volumes/

There's a .dockerignore file for mapped volumes to not map specific file patterns kinda like .gitignore

Reply:
And I'd personally put the "user facing" bash scripts in package json scripts section usuaully so no direct bash script calling.

U can get rid of environment checks to see if ur in docker in ur bash script too since you're guaranteeing that it will be called through docker exec through "npm run <script>"

I would add a comment to those bash scripts that they are meant to be run inside a container or put them somewhere close to the dockerfile to imply it.

Reply:
>  you're guaranteeing that it will be called through docker exec through "npm run <script>"

That won't always be the case, I would like to dump and restore Hasura from my host system.

> U can use this to run any command inside the container. Then u can call ur bash script to setup the db.

Yes I was aware of that, I do use `docker exec` when running some database scripts. 

--- 

I  believe I'm still a bit confused on what your suggested goal is :sweat_smile: 

Did you mean:  not having a `migration` service? having a separate image using `timescale/timescaledb:latest-pg16` as base? running `npm run setup` manually after `docker-compose up`?

If you're up to you could let me know more on slack!
Alternatively you could also make a PR on this branch :smile:

Reply:
Let's go on slack 😄